management-ui/server.py

```diff
@@ -58,6 +85,19 @@ async def get_session(config: Configuration) -> M1Session:
                                        config.get('certificate_signing_class'))
     return _m1_session
 
+async def initialize_media_configuration(session: M1Session) -> MediaConfiguration:
+    global _media_configuration
+    data_store_dir = config.get('data_store')
+    if data_store_dir is not None:
+        data_store = await JSONFileDataStore(config.get('data_store'))
+    else:
+        data_store = None
+    if _media_configuration is None:
+        _media_configuration = await MediaConfiguration(persistent_data_store=data_store)
```

Comment:
You can get MediaConfiguration to use your M1Session object for its synchronisation by providing it in the *m1_session* argument to the MediaConfiguration initialisation. So this line would become:
```python
_media_configuration = await MediaConfiguration(persistent_data_store=data_store, m1_session=session)
```